sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_async_lro.py

```diff
@@ -0,0 +1,89 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+from urllib.parse import urlencode
+from azure.core.polling.base_polling import OperationFailed, BadStatus
+from azure.core.polling.async_base_polling import AsyncLROBasePolling
+
+
+_FINISHED = frozenset(["succeeded", "cancelled", "failed", "partiallysucceeded"])
+_FAILED = frozenset(["failed"])
+_SUCCEEDED = frozenset(["succeeded", "partiallysucceeded"])
+
+
+class TextAnalyticsAsyncLROPoller(AsyncLROBasePolling):
+    def __init__(self,
+        timeout=30,
+        lro_algorithms=None,
+        lro_options=None,
+        path_format_arguments=None,
+        **operation_config
+    ):
+
+        super(TextAnalyticsAsyncLROPoller, self).__init__(
+            timeout=timeout,
+            lro_algorithms=lro_algorithms,
+            lro_options=lro_options,
+            path_format_arguments=path_format_arguments,
+            **operation_config
+        )
+
+    def finished(self):
+        """Is this polling finished?
+        :rtype: bool
+        """
+        return self._finished(self.status())
+
+    def _finished(self, status):
+        if hasattr(status, "value"):
+            status = status.value
+        return str(status).lower() in _FINISHED
+
+    def _failed(self, status):
+        if hasattr(status, "value"):
+            status = status.value
+        return str(status).lower() in _FAILED
+
+    def _raise_if_bad_http_status_and_method(self, response):
+        # type: (ResponseType) -> None
+        """Check response status code is valid.
+
+        Must be 200, 201, 202, or 204.
+
+        :raises: BadStatus if invalid status.
+        """
+        code = response.status_code
+        if code in {200, 201, 202, 204}:
+            return
+        raise BadStatus(
+            "Invalid return status {!r} for {!r} operation".format(
+                code, response.request.method
+            )
+        )
+
+    async def _poll(self):  # pylint:disable=invalid-overridden-method
```

Comment:
Does this method have the only customization for the polling method? In general we'd rather subclass the Polling types rather than the BasePolling directly. Wondering if we can instead subclass [StatusPolling](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/polling/base_polling.py#L324), do customization there, and pass this in as an [lro_algorithm](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py#L427). 

This feedback is not for this release, just an FYI

Reply:
The main reason I did this was to add some states for checking in the result from the service.  We have a few inconsistencies, such as new states ("partiallysucceeded", "cancelling") and states with an alternate spelling ("cancelled" instead of "canceled").  Since the states are checked in the body of _poll I wasn't sure how else to ensure that the additional states could be accounted for.  I have also subclassed OperationResourcePolling but for different reasons.

Reply:
Got it - makes sense, didn't see that before. Thanks!